src/Bass/Shared/ChannelReferences.cs

```diff
@@ -20,6 +21,10 @@ public static class ChannelReferences
         public static void Add(int Handle, int SpecificHandle, object proc)
         {
 #if !__IOS__
+            // in .NET iOS, the __IOS__ constant cannot be seen, so rely on RuntimeFeature instead.
+            if (!RuntimeFeature.IsDynamicCodeCompiled)
```

Comment:
Any reason for using `IsDynamicCodeCompiled` over `IsDynamicCodeSupported`? It looks like the latter is intended for this purpose, as documented in the PR: https://github.com/dotnet/runtime/issues/25959

Reply:
I've checked back on this once again as I've got very confusing results over these two booleans the last time, and back then I didn't quite understand how Mono interpreter plays with it.

| Configurations | `RuntimeFeature.IsDynamicCodeCompiled` | `RuntimeFeature.IsDynamicCodeSupported` |
|----------------|----------------------------------------|-------------------------------------------------------------------|
| Debug iPhone | `false` | `false` (!) |
| Release iPhone | `false` | `false` |
| Debug iPhone + `UseInterpreter` | `false` | `true` |
| Release iPhone + `UseInterpreter` | `false` | `true` |
| Debug iPhone + `MtouchInterpreter=-all` | `false` | `true` |
| Release iPhone + `MtouchInterpreter=-all` | `false` | `true` |
| Debug Simulator | `true` | `true` |
| Release Simulator | `true` | `true` |
| Debug Simulator + `UseInterpreter` | `false` | `true` |
| Release Simulator + `UseInterpreter ` | `false` | `true` |

(simulator + `MtouchInterpreter` crashes AOT compiler)

The results of the "Debug iPhone" configuration shocked me a little as the app is built with interpreter, but it's good that it behaves that way as opposed to showing different results compared to release mode.

Anyway the results make sense now, I'll update accordingly.